src/analytics-dashboard/analytics-dashboard.controller.ts

```diff
@@ -0,0 +1,67 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete, UseGuards, UnauthorizedException } from "@nestjs/common";
+import { AnalyticsDashboardService } from "./analytics-dashboard.service";
+import { AuthGuard } from "@nestjs/passport";
+import { ApiTags, ApiBearerAuth, ApiResponse } from "@nestjs/swagger";
+import { User } from "src/decorators/User";
+import { IUser } from "src/auth/User";
+import {
+  CategoryWiseStatsDto,
+  CourseWiseStatsDto,
+  DepartmentWiseStatsDto,
+  GenderWiseStatsDto,
+  StatsDto,
+} from "./dto/get.dto";
+
+@Controller("analytics-dashboard")
+@UseGuards(AuthGuard("jwt"))
+@ApiTags("analytics-dashboard")
+@ApiBearerAuth("jwt")
+export class AnalyticsDashboardController {
+  constructor(private readonly analyticsDashboardService: AnalyticsDashboardService) {}
+
+  @Get("statsOverall/:seasonId")
+  @ApiResponse({ type: StatsDto })
+  async getPlacementStatisticsOverall(@Param("seasonId") seasonId: string, @User() user: IUser) {
+    if (user.role !== "ADMIN") {
```

Comment:
it would be better if we have another auth guard for admin and use it here.

Reply:
another auth guard?

Reply:
yes

Reply:
like we have for jwt we can have another that extends it and also checks for role== admin